include/deal.II/grid/grid_generator.h

```diff
@@ -388,6 +433,37 @@ namespace GridGenerator
                   const double        half_length = 1.0);
 
   /**
+  * \brief A center cell with stacks of cell protruding from each surface.
+  *
+  * Each of the square mesh cells is Cartesian and has size one in each
+  * coordinate direction. The center of cell number zero is the
+  * origin.
+  *
+  * The legs of the cross are stacked on the faces of the center cell,
+  * in the usual order of deal.II cells, namely first $-x$, then $x$,
+  * then $-y$ and so on. The corresponding entries in
+  * <code>sizes</code> name the number of cells stacked on this
+  * face. All numbers may be zero, thus L- and T-shaped domains are
```

Comment:
Can you add a statement that this means that `sizes` must have size 2*dim (==`GeometryInfo<dim>::faces_per_cell`)?